opts.go

```diff
@@ -162,6 +166,50 @@ func makeOptValueHandling(h configHandling) Option {
 	}
 }
 
+var (
+	// FieldMergeValues option configures all merging and unpacking operations to use
+	// the default merging behavior for the specified field. This overrides the any struct
+	// tags during unpack for the field. Nested field names can be defined using dot
+	// notation.
+	FieldMergeValues = makeFieldOptValueHandling(cfgMergeValues)
+
+	// FieldReplaceValues option configures all merging and unpacking operations to
+	// replace old dictionaries and arrays while merging for the specified field. This
+	// overrides the any struct tags during unpack for the field. Nested field names
+	// can be defined using dot notation.
+	FieldReplaceValues = makeFieldOptValueHandling(cfgReplaceValue)
+
+	// FieldAppendValues option configures all merging and unpacking operations to
+	// merge dictionaries and append arrays to existing arrays while merging for the
+	// specified field. This overrides the any struct tags during unpack for the field.
+	// Nested field names can be defined using dot notation.
+	FieldAppendValues = makeFieldOptValueHandling(cfgArrAppend)
+
+	// FieldPrependValues option configures all merging and unpacking operations to
+	// merge dictionaries and prepend arrays to existing arrays while merging for the
+	// specified field. This overrides the any struct tags during unpack for the field.
+	// Nested field names can be defined using dot notation.
+	FieldPrependValues = makeFieldOptValueHandling(cfgArrPrepend)
+)
+
+func makeFieldOptValueHandling(h configHandling) func(...string) Option {
+	return func(fieldName ...string) Option {
+		return func(o *options) {
+			if o.fieldValueHandling == nil {
+				o.fieldValueHandling = make(map[string]configHandling)
+			}
+			for _, name := range fieldName {
+				// field value config options are rendered into a Config; the '*' represents the handling method
+				// for everything nested under this field.
+				if !strings.HasSuffix(name, ".*") {
+					name = fmt.Sprintf("%s.*", name)
+				}
+				o.fieldValueHandling[name] = h
+			}
+		}
+	}
+}
```

Comment:
We can update this function, to reduce allocation per call (presuming we reuse the options), and remove `fieldValueHandling` as temporary variable by rewriting `makeFieldOptValueHandling` like this:

```
func makeFieldOptValueHandling(h configHandling) func(...string) Option {
	return func(fieldName ...string) Option {
		if len(fieldName) == 0 {
			return func(_ *options) {}
		}

		table := make(map[string]configHandler, len(fieldName))
		for _, name := range fieldName {
			// for everything nested under this field.
			if !strings.HasSuffix(name, ".*") {
				name = fmt.Sprintf("%s.*", name)
			}
			table[name] = h
		}

		return func(o *options) {
			if o.fieldValueHandlingConfig == nil {
				o.fieldValueHandlingConfig = New()
			}
			o.fieldValueHandlingConfig.Merge(table, PathSep(o.pathSep))
		}
	}
}
```

Reply:
Okay will update to use this.